### PR TITLE
Prepare for release 0.2.3

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This CL preps for the 0.2.3 release and pulls in @bkonyi's changes from https://github.com/flutter/devtools/pull/1692.

### CODE REVIEW

File: packages/devtools_app/lib/src/debugger/debugger_state.dart [hunk 1]
```diff
@@ -84,11 +84,11 @@ class DebuggerState {
     }
   }
 
-  Future<Success> pause() => _service.pause(isolateRef.id);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/debugger_state.dart]: why are these types removed? Were they incorrect?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/debugger_state.dart]: yes vm_service 3.0.0 included a breaking change. The type is now dynamic as a Sentinel can be returned

Comment (bkonyi) [reply packages/devtools_app/lib/src/debugger/debugger_state.dart]: Any RPC that requires an isolateId parameter can return a Sentinel if the isolate no longer exists. The spec didn't explicitly state this even though this has always been true, so it was possible to run into runtime type errors if timing was bad.

File: packages/devtools_app/lib/src/debugger/debugger_state.dart [hunk 2]
```diff
@@ -118,27 +118,27 @@ class DebuggerState {
...
     return _service.removeBreakpoint(isolateRef.id, breakpoint.id);
   }
 
-  Future<void> setExceptionPauseMode(String mode) {
+  Future<dynamic> setExceptionPauseMode(String mode) {
     return _service.setExceptionPauseMode(isolateRef.id, mode);
   }
 
-  Future<Stack> getStack() {
+  Future<dynamic> getStack() {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/debugger_state.dart]: this should continue to return Future<Stack> 
We should return null or an empty Stack object if a Sentinel is returned. Code using this API is not expecting Sentinel objects and we are hiding the bug by returning dynamic. Alternately we could return Object here and elsewhere and then fix all the upstream analysis errors. Likely it is simpler to fix them all up within the State objects. users of the State object shouldn't need to know about the VM Sentinel concept.

File: packages/devtools_app/lib/src/memory/memory_controller.dart [hunk 3]
```diff
@@ -143,7 +145,7 @@ class MemoryController {
   List<String> sortLibrariesByNormalizedNames() =>
       libraryCollection.librarires.keys.toList()..sort();
 
-  Future getObject(String objectRef) async =>
+  Future<dynamic> getObject(String objectRef) async =>
```
Comment (jacob314) [inline packages/devtools_app/lib/src/memory/memory_controller.dart]: make this Future<Object>

File: packages/devtools_app/lib/src/profiler/cpu_profile_service.dart [hunk 4]
```diff
@@ -30,7 +30,7 @@ class CpuProfilerService {
   ValueNotifier<Flag> get profileGranularityFlagNotifier =>
       serviceManager.vmFlagManager.flag(vm_flags.profilePeriod);
 
-  Future<Success> clearCpuSamples() {
+  Future<dynamic> clearCpuSamples() {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/profiler/cpu_profile_service.dart]: For cases like this where we don't really care about the return value default to just returning
Future<void>

File: packages/devtools_app/lib/src/vm_service_wrapper.dart [hunk 5]
```diff
@@ -151,7 +151,7 @@ class VmServiceWrapper implements VmService {
   void dispose() => _vmService.dispose();
 
   @override
-  Future evaluate(
+  Future<dynamic> evaluate(
```
Comment (jacob314) [inline packages/devtools_app/lib/src/vm_service_wrapper.dart]: Prefer Future<Object> if the type system allows it.

File: packages/devtools_app/pubspec.yaml [hunk 6]
```diff
@@ -64,11 +64,11 @@ dev_dependencies:
   build_web_compilers: '>=2.6.2 <3.0.0'
   matcher: ^0.12.3
   mockito: ^4.0.0
-  test: ^1.0.0
+  test: any
```
Comment (jacob314) [inline packages/devtools_app/pubspec.yaml]: comment that this is pinned by Flutter so we don't care.

File: packages/devtools_app/lib/src/memory/memory_controller.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/memory/memory_controller.dart]: done

File: packages/devtools_app/lib/src/profiler/cpu_profile_service.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/profiler/cpu_profile_service.dart]: done

File: packages/devtools_app/lib/src/vm_service_wrapper.dart (hunk 5 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/vm_service_wrapper.dart]: done

File: packages/devtools_app/pubspec.yaml (hunk 6 above)
Comment (kenzieschmoll) [reply packages/devtools_app/pubspec.yaml]: done

File: packages/devtools_app/lib/src/debugger/debugger_state.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/debugger/debugger_state.dart]: done

**Pull request merged**